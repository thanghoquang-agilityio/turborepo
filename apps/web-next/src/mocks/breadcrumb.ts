export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
