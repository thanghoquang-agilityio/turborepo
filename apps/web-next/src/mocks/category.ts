export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
