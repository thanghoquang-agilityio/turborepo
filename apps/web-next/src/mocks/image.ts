import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants'

export const MOCK_IMAGE = {
  data: {
    id: '24',
    attributes: {
      name: 'default',
      url: SRC_IMAGE_NOT_AVAILABLE,
    },
  },
}

export const MOCK_LIST_IMAGE_RESPONSE = [
  {
    name: 'default1',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default2',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default3',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default4',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
]
