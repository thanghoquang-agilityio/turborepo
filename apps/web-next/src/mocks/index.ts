export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
