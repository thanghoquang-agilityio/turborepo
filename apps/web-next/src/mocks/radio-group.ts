export const MOCK_COLOR_RADIO_GROUP = [
  { color: 'yellow green' },
  { color: 'violet' },
  { color: 'wheat' },
  { color: 'slate blue' },
  { color: 'salmon - dark green' },
]

export const MOCK_SIZE_RADIO_GROUP = [
  { size: 'Small', stock: 4 },
  { size: 'Medium', stock: 4 },
  { size: 'Large', stock: 0 },
]
