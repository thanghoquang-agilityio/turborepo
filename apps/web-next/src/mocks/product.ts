import { ProductVariantModel } from '@/types'

import { MOCK_IMAGE } from './image'

// New flattened product variants for testing
export const MOCK_PRODUCT_VARIANTS: ProductVariantModel[] = [
  {
    id: 1,
    color: 'slate blue',
    size: 'small,medium,large,extra large,XXL',
    stock: 20,
    tax: 5,
  },
  {
    id: 2,
    color: 'blue',
    size: 'small,medium,large,extra large,XXL',
    stock: 20,
    tax: 5,
  },
]

export const MOCK_PRODUCT = {
  id: '1',
  image: MOCK_IMAGE,
  name: 'TDX Sinkers',
  variants: 5,
  price: 675.0,
  star: 5,
  comments: 121,
  description: '',
  discount: 0,
  likes: 11,
  brand: 'MOCK BRAND',
  productVariants: {
    data: MOCK_PRODUCT_VARIANTS,
  },
}

export const MOCK_PRODUCTS_RESPONSE = [
  {
    id: '1',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '2',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '3',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '4',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: MOCK_PRODUCT_VARIANTS,
      },
      image: MOCK_IMAGE,
    },
  },
]
