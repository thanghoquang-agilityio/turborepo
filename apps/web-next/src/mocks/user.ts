import { UserModel } from '@/types'

import { MOCK_IMAGE } from './image'

export const MOCK_USER: UserModel = {
  username: 'thanghq',
  email: 'thanghq@gmail.com',
  image: MOCK_IMAGE,
  cartItems: [
    {
      id: 3,
      size: 'large',
      quantity: 2,
    },
    {
      id: 4,
      size: 'small',
      quantity: 2,
    },
    {
      id: 5,
      size: 'medium',
      quantity: 3,
    },
  ],
}
