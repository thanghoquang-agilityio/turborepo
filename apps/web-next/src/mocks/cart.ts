import { CartItemResponse, LegacyCartItemResponse } from '@/types'
import { MOCK_PRODUCT_VARIANTS } from './product'

export const MOCK_CART_ITEM = {
  id: '1',
  variantId: '1',
  name: 'Embrace Sideboard',
  color: 'wheat',
  size: 'small',
  quantityDefault: 2,
  stock: 4,
  price: 53,
  star: 1,
  comments: 1,
  tax: 1,
  discount: 1,
}

// New flattened cart response structure
export const MOCK_CART_RESPONSE: CartItemResponse[] = [
  {
    id: 1,
    documentId: 'cart-item-1',
    size: 'small',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[0],
    userId: 1,
  },
  {
    id: 2,
    documentId: 'cart-item-2',
    size: 'large',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[1],
    userId: 1,
  },
]

// Legacy cart response for backward compatibility
export const MOCK_CART_RESPONSE_LEGACY: LegacyCartItemResponse[] = [
  {
    id: '1',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '2,10,0,23,20',
            tax: 12,
          },
        },
      },
      size: 'small',
      quantity: 2,
    },
  },
  {
    id: '2',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '20,10,20,23,20',
            tax: 12,
          },
        },
      },
      size: 'large',
      quantity: 2,
    },
  },
]

export const MOCK_ORDER_SUMMARY = {
  subTotal: 396,
  itemTotal: 3,
}
