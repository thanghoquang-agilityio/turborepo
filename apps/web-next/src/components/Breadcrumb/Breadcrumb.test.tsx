import { render } from '@testing-library/react'

import { MOCK_BREADCRUMB_ITEMS } from '@/mocks'

import { Breadcrumb } from '.'

describe('Breadcrumb', () => {
  beforeEach(() => {
    const mockIntersectionObserver = jest.fn()
    mockIntersectionObserver.mockReturnValue({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    })
    window.IntersectionObserver = mockIntersectionObserver
  })

  it('Renders match snapshot', () => {
    const { container } = render(<Breadcrumb items={MOCK_BREADCRUMB_ITEMS} />)
    expect(container).toMatchSnapshot()
  })
})
