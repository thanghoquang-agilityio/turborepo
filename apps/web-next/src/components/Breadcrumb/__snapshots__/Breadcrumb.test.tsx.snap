// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Renders match snapshot 1`] = `
<div>
  <div>
    <nav
      aria-label="Breadcrumbs"
      class="bg-background-300 w-fit rounded-medium"
      data-slot="base"
    >
      <ol
        class="flex flex-wrap list-none rounded-small bg-default-100 max-w-fit px-2.5 py-1.5 gap-3.5"
        data-slot="list"
      >
        <li
          class="flex items-center gap-3.5"
          data-slot="base"
          href="/"
        >
          <a
            class="flex gap-1 items-center cursor-pointer whitespace-nowrap line-clamp-1 tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small active:opacity-disabled transition-opacity no-underline text-secondary-500 hover:text-primary-300 hover:opacity-1 data-[current=true]:text-primary-300 data-[current=true]:font-semibold"
            data-slot="item"
            href="/"
          >
            Dashboard
          </a>
          <span
            aria-hidden="true"
            class="px-1 text-secondary-500"
            data-slot="separator"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1em"
              role="presentation"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </span>
        </li>
        <li
          class="flex items-center gap-3.5"
          data-slot="base"
          href="products"
        >
          <a
            class="flex gap-1 items-center cursor-pointer whitespace-nowrap line-clamp-1 tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small active:opacity-disabled transition-opacity no-underline text-secondary-500 hover:text-primary-300 hover:opacity-1 data-[current=true]:text-primary-300 data-[current=true]:font-semibold"
            data-slot="item"
            href="products"
          >
            Products
          </a>
          <span
            aria-hidden="true"
            class="px-1 text-secondary-500"
            data-slot="separator"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1em"
              role="presentation"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </span>
        </li>
        <li
          class="flex items-center gap-3.5"
          data-slot="base"
          href="7"
        >
          <span
            aria-current="page"
            aria-disabled="true"
            class="flex gap-1 items-center whitespace-nowrap line-clamp-1 tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small no-underline cursor-default transition-opacity text-secondary-500 hover:text-primary-300 hover:opacity-1 data-[current=true]:text-primary-300 data-[current=true]:font-semibold"
            data-current="true"
            data-slot="item"
            href="7"
            role="link"
          >
            Detail Product
          </span>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;
